module/scheduler.cpp

```diff
@@ -328,25 +328,31 @@ void Scheduler::HandleResponseFromWorker(const internal::WorkerResponse& res) {
   if (txn->procedure_case() == Transaction::ProcedureCase::kNewMaster) {
     auto key = txn->write_set().begin()->first;
     auto counter = txn->internal().master_metadata().at(key).master() + 1;
-    auto remaster_result = remaster_manager_.RemasterOccured(key, counter);
+    ProcessRemasterResult(remaster_manager_.RemasterOccured(key, counter));
+  }
 
-    for (auto unblocked_txn_holder : remaster_result.unblocked) {
-      SendToLockManager(unblocked_txn_holder);
-    }
-    for (auto unblocked_txn_holder : remaster_result.should_abort) {
-      AbortTransaction(unblocked_txn_holder);
-    }
+  switch (txn->status()) {
+    case TransactionStatus::COMMITTED:
+      SendToCoordinatingServer(txn_holder);
+      break;
+    case TransactionStatus::ABORTED:
+      AbortTransaction(txn_holder, true);
+      break;
+    default:
+      LOG(ERROR) << "Invalid txn status from worker: " << txn->status();
   }
+}
 
+void Scheduler::SendToCoordinatingServer(TransactionHolder* txn_holder) {
```

Comment:
This method should take in a txn_id instead of a holder. 
Removing the transaction from `all_txns_` in the end implies that the holder must come from that map. So it is safer to get the holder from that map at the beginning via a txn_id instead of relying on the method user to provide the correct holder.